tests/integration/lxd/test_launcher.py

```diff
@@ -573,3 +573,44 @@ def test_launch_instance_config_incompatible(core20_instance):
 
     assert core20_instance.exists()
     assert core20_instance.is_running()
+
+
+def test_launch_instance_id_map_incompatible(core20_instance):
+    """Raise an error if the instance's id map is incompatible.
+    If auto_clean is true, delete and recreate the instance.
+    """
+    base_configuration = bases.BuilddBase(alias=bases.BuilddBaseAlias.FOCAL)
+
+    lxc = lxd.LXC()
+    lxc.config_set(
+        instance_name=core20_instance.instance_name,
+        key="raw.idmap",
+        value="",  # equivalent to `lxc config unset`, which is not yet implemented
+    )
+
+    # will raise compatibility error when auto_clean is false
+    with pytest.raises(bases.BaseCompatibilityError) as exc_info:
+        lxd.launch(
+            name=core20_instance.name,
+            base_configuration=base_configuration,
+            image_name="20.04",
+            image_remote="ubuntu",
+            map_user_uid=True,
+        )
+
+    assert exc_info.value.brief == (
+        "Incompatible base detected: "
+        "the instance's id map ('raw.idmap') is not configured as expected."
+    )
+
```

Comment:
The bit below this and the bit above seem like two separate tests.

Reply:
Yep.  The reason was `copypasta`, one of my common programming errors!

I split this apart and added a TODO for the tests I copied from.